miniqiita.go

```diff
@@ -74,10 +74,8 @@ func (c *Client) GetUserItems(ctx context.Context, userID string, page, perPage
 }
 
 func (c *Client) newRequest(ctx context.Context, method, relativePath string, queries, headers map[string]string, reqBody io.Reader) (*http.Request, error) {
-	reqURL := *c.BaseURL
-
 	// set path
-	reqURL.Path = path.Join(reqURL.Path, relativePath)
+	reqURL, err := c.BaseURL.Parse(relativePath)
```

Comment:
This is the code comment on `Parse()`.
```go
// Parse parses a URL in the context of the receiver. The provided URL
// may be relative or absolute. Parse returns nil, err on parse
// failure, otherwise its return value is the same as ResolveReference.
func (u *URL) Parse(ref string) (*URL, error) {
	refurl, err := Parse(ref)
	if err != nil {
		return nil, err
	}
	return u.ResolveReference(refurl), nil
}
```